File: trunk/src/ch/usi/inf/genesis/data/repository/InFamixWrapper.java
package ch.usi.inf.genesis.data.repository;
import java.io.File;
import java.io.IOException;


public class InFamixWrapper implements IExternalModelParserWrapper{

	private final String famixPath;
	private final FamixLanguage language;

	public InFamixWrapper(final String famixPath, final FamixLanguage language){
		this.famixPath = famixPath;
		this.language = language;
	}

	public File execute(final String sourceFilePath, final String outputFilePath, final boolean asynchronous) throws IOException{
		final File source = new File(sourceFilePath);
		final File output = new File(outputFilePath);

		System.out.println("Source: " + source.getAbsolutePath() + "\nOutput: " + output.getAbsolutePath());
		return this.execute(source, output, asynchronous);
	}

	public File execute(final File sourceFile, final File outputFileName, final boolean asynchronous) throws IOException{
		//Check if the mse output folder exists. In case it is missing, it is created.
		if(!outputFileName.exists())
			outputFileName.mkdirs();
		
		final Process p = Runtime.getRuntime().exec(String.format("%s -lang %s -path %s -mse %s", famixPath, language.getId(), sourceFile.getAbsolutePath(), outputFileName.getAbsolutePath()));
		if(!asynchronous){
			try {
				p.waitFor();
			} catch (final InterruptedException e) {
				e.printStackTrace();
			}
		}
		//Delete Workspace
		new File(famixPath.substring(0,famixPath.lastIndexOf("/"))).delete();
		return outputFileName;
	}

	public String getFamixPath() {
		return famixPath;
	}

	public FamixLanguage getLanguage() {
		return language;
	}
}

File: trunk/src/scala/ch/usi/genesis/data/repository/InFamixWrapper.scala
package ch.usi.genesis.data.repository

import java.io.{BufferedReader, InputStreamReader, InputStream, File}


class InFamixWrapper(infamixPath : String, lang : String) extends ExternalParserWrapper{

  val bashScriptPath = "utils/mse.sh"

  def execute(sourceFilePath: String, outputFile: String, synchronous: Boolean): File = {
                  //script usage: <infamix_path> <infamix_lang> <project_path> <mse_path>"
    val cmd = "./" +bashScriptPath+" "+infamixPath+" "+lang + " " + sourceFilePath + " " + outputFile
    val p: Process = Runtime.getRuntime.exec(cmd)

    var stderr: InputStream = p.getInputStream
    var isr: InputStreamReader = new InputStreamReader(stderr)
    var br: BufferedReader = new BufferedReader(isr)
    var line: String = null

    line = br.readLine
//    while(line != null){
//      println(line)
//      line = br.readLine
//    }
    println("InFamix Done.")

    try {
      println("Waiting...")
      p.waitFor()
    }
    catch {
      case e: InterruptedException => {
        e.printStackTrace()
      }
    }

   new File(outputFile)
  }

  override def execute(sourceFile: File, outputFileName: File, synchronous: Boolean): File = {
    null
  }
}